File: python/demos/demo_teb.py
#!/usr/bin/env python3
"""TEB demo."""

from __future__ import annotations

from demo_common import run_local

# Imported from the submodule directly, mirroring demo_dwa.py's precedent (the
# package re-export lives in navigation/local_planning/band/__init__.py).
from navigation.local_planning.band.teb import TebPlanner

if __name__ == "__main__":
    run_local("teb", TebPlanner)

File: cpp/demos/demo_teb.cpp
#include "demo_common.hpp"
#include "navigation/local_planning/band/teb.hpp"

int main(int argc, char** argv) {
  try {
    return demo::run_local(argc, argv, "teb", [](const navigation::core::ParamSet& p) {
      return navigation::local_planning::TebPlanner(p);
    });
  } catch (const std::exception& e) {
    std::cerr << "error: " << e.what() << "\n";
    return 1;
  }
}
